PWGHF/Tasks/taskMcEfficiency.cxx

```diff
@@ -0,0 +1,290 @@
+    std::vector<bool> hasITS(mcParticles.size(), false);
+    std::vector<bool> hasTPC(mcParticles.size(), false);
+    std::vector<bool> selected(mcParticles.size(), false);
+    for (auto& track : tracks) {
+      if (track.mcParticleId() >= 0) {
+        tracked[track.mcParticleId()] = true;
+        if (checkTrack(track)) {
+          selected[track.mcParticleId()] = true;
+        }
+        if (track.hasITS()) {
+          hasITS[track.mcParticleId()] = true;
+        }
+        if (track.hasTPC()) {
+          hasTPC[track.mcParticleId()] = true;
+        }
+      }
+    }
+
+    for (auto pdgCode : (std::vector<int>)pdgCodes) {
+      auto mass = RecoDecay::getMassPDG(pdgCode);
+
+      for (auto& mcParticle : mcParticles) {
+        if (mcParticle.pdgCode() != pdgCode) {
+          continue;
+        }
+        hCandidates->Fill(kHFStepMC, mcParticle.pt(), mass, pdgCode, 1.0, true);
+
+        if (mcParticle.daughtersIds().size() != 2) {
+          LOGP(fatal, "Invalid numbers of daughters for D0(bar) {}: {}", mcParticle.globalIndex(), mcParticle.daughtersIds().size());
+        }
+
+        auto prong0Id = mcParticle.daughtersIds()[0];
+        auto prong1Id = mcParticle.daughtersIds()[1];
+
+        if (prong0Id < 0 || prong1Id < 0) {
+          continue;
+        }
+
+        bool inAcceptance = true;
+        for (auto& daugther : mcParticle.daughters_as<aod::McParticles>()) {
+          if (daugther.pt() < mcAcceptancePt || std::abs(daugther.eta()) > mcAcceptanceEta) {
+            inAcceptance = false;
+          }
+        }
+
+        if (inAcceptance) {
+          hCandidates->Fill(kHFStepAcceptance, mcParticle.pt(), mass, pdgCode, 1.0, true);
+        }
+
+        if (tracked[prong0Id] && tracked[prong1Id]) {
+          hCandidates->Fill(kHFStepTrackable, mcParticle.pt(), mass, pdgCode, 1.0, true);
+          if (inAcceptance) {
+            hCandidates->Fill(kHFStepAcceptanceTrackable, mcParticle.pt(), mass, pdgCode, 1.0, true);
+          } else {
+            LOGP(debug, "Candidate {} not in acceptance but tracked.", mcParticle.globalIndex());
+            for (auto& daugther : mcParticle.daughters_as<aod::McParticles>()) {
+              LOGP(debug, "   MC: pt={} eta={}", daugther.pt(), daugther.eta());
+            }
+          }
+          for (auto& daugther : mcParticle.daughters_as<aod::McParticles>()) {
+            hTrackablePtEta->Fill(kTrackableAll, daugther.pt(), daugther.eta());
+            if (hasITS[daugther.globalIndex()]) {
+              hTrackablePtEta->Fill(kTrackableITS, daugther.pt(), daugther.eta());
+            }
+            if (hasTPC[daugther.globalIndex()]) {
+              hTrackablePtEta->Fill(kTrackableTPC, daugther.pt(), daugther.eta());
+            }
+            if (hasITS[daugther.globalIndex()] && hasTPC[daugther.globalIndex()]) {
+              hTrackablePtEta->Fill(kTrackableITSTPC, daugther.pt(), daugther.eta());
+            }
```

Comment:
`pt`, `eta`, `globalIndex` can be stored

Reply:
But those are expression columns (so no recalculation). Is it ok to keep them?